File: Examples/JXPagerViewExample-OC/JXPagerViewExample-OC/Example/Common/BaseNavigationController.m
//
//  BaseNavigationController.m
//  JXPagerViewExample-OC
//
//  Created by jiaxin on 2019/3/15.
//  Copyright © 2019 jiaxin. All rights reserved.
//

#import "BaseNavigationController.h"

@interface BaseNavigationController () <UIGestureRecognizerDelegate>

@end

@implementation BaseNavigationController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.interactivePopGestureRecognizer.delegate = self;
    
    if (@available(iOS 13.0, *)) {
        // iOS15 系统下 push 后导航条变黑色 且 有卡顿
        UINavigationBarAppearance *barAppearance = [[UINavigationBarAppearance alloc] init];
        barAppearance.backgroundColor = [UIColor whiteColor];
        self.navigationBar.scrollEdgeAppearance = barAppearance;
        self.navigationBar.standardAppearance = barAppearance;
    } else {
        // Fallback on earlier versions
    }
    
}

#pragma mark - UIGestureRecognizerDelegate

- (BOOL)gestureRecognizerShouldBegin:(UIGestureRecognizer *)gestureRecognizer {
    if (self.viewControllers.count == 1) {
        return NO;
    }
    return YES;
}

- (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
    return YES;
}

@end

File: Examples/JXPagingViewExample/JXPagingViewExample/Example/Common/BaseNavigationController.swift
//
//  BaseNavigationController.swift
//  JXPagingView
//
//  Created by jiaxin on 2018/8/28.
//  Copyright © 2018年 jiaxin. All rights reserved.
//

import UIKit

class BaseNavigationController: UINavigationController {

    override func viewDidLoad() {
        super.viewDidLoad()

        self.interactivePopGestureRecognizer?.delegate = self
        
        if #available(iOS 13.0, *) {
            // iOS15 系统下 push 后导航条变黑色 且 有卡顿
            let barAppearance = UINavigationBarAppearance()
            barAppearance.backgroundColor = .white
            self.navigationBar.scrollEdgeAppearance = barAppearance
            self.navigationBar.standardAppearance = barAppearance;
        }
    }
}

extension BaseNavigationController: UIGestureRecognizerDelegate {
    func gestureRecognizerShouldBegin(_ gestureRecognizer: UIGestureRecognizer) -> Bool {
        if self.viewControllers.count == 1 {
            return false
        }
        return true
    }

    func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
        return false
    }
}
